dev/tools/dartdoc.dart

```diff
@@ -56,24 +63,61 @@ dependencies:
     '--use-categories'
   ];
 
-  for (String libraryRef in _libraryRefs()) {
+  for (String libraryRef in libraryRefs()) {
     String name = path.basename(libraryRef);
     args.add('--include-external');
     args.add(name.substring(0, name.length - 5));
   }
 
-  _findSkyServicesLibraryNames().forEach((String libName) {
+  findSkyServicesLibraryNames().forEach((String libName) {
     args.add('--include-external');
     args.add(libName);
   });
 
   process = await Process.start('pub', args, workingDirectory: 'dev/docs');
-  _print(process.stdout);
-  _print(process.stderr);
-  exit(await process.exitCode);
+  printStream(process.stdout);
+  printStream(process.stderr);
+  int exitCode = await process.exitCode;
+
+  if (exitCode != 0)
+    exit(exitCode);
+
+  createIndexAndCleanup();
+}
+
+/// Creates a custom index.html because we try to maintain old
+/// paths. Cleanup unused index.html files no longer needed.
+void createIndexAndCleanup() {
+  print('\nCreating a custom index.html in $DOC_ROOT/index.html');
+  renameApiDir();
+  copyIndexToRootOfDocs();
+  addHtmlBaseToIndex();
+  putRedirectInOldIndexLocation();
+  print('\nDocs ready to go!');
+}
+
+Directory renameApiDir() {
```

Comment:
We're not using the return values of `renameApiDir` (or `copyIndexToRootOfDocs` below).